### PR TITLE
Snapshots

### PR DESCRIPTION
**Pull request by prolic**

### CODE REVIEW

**Comment by codeliner** [comment]
@prolic Looks great so far :+1:

File: src/Aggregate/AggregateRepository.php [hunk 1]
```diff
@@ -207,4 +226,33 @@ public function getAggregateRoot($aggregateId)
...
+        
+        $aggregateRoot = $snapshot->getAggregateRoot();
+
+        $streamEvents = $this->streamStrategy->read(
+            $this->aggregateType,
+            $aggregateId,
+            $snapshot->getLastVersion() + 1
+        );
+
+        if (count($streamEvents) === 0) {
```
Comment (prolic) [inline src/Aggregate/AggregateRepository.php]: remove!

File: src/Aggregate/AggregateRepository.php [hunk 2]
```diff
@@ -190,6 +199,16 @@ public function getAggregateRoot($aggregateId)
...
+            if ($aggregateRoot) {
+                $this->identityMap->add($this->aggregateType, $aggregateId, $aggregateRoot);
+
+                return $aggregateRoot;
+            }
+        }
+
         $streamEvents = $this->streamStrategy->read($this->aggregateType, $aggregateId);
 
         if (count($streamEvents) === 0) {
```
Comment (prolic) [inline src/Aggregate/AggregateRepository.php]: remove!

**Comment by prolic** [comment]
@codeliner ready!

**Comment by codeliner** [comment]
@prolic great work :+1: . This will kill discussions about replaying events takes to much time :smile:

**Pull request merged**